### ISSUE TITLE
Update error on Mac OS High Sierra without codesign

### ISSUE THREAD

**Issue by Dave-K9** [sparkle-project/Sparkle#1239]
I tried the latest version of Sparkle (19.1) on Mac OS (10.13.4) using DSA keys but without Apple signing, which is supposed to work according to the docs.  I followed the online documentation to the letter, but I end up getting the dreaded message:  

      "Update Error! An error occurred in retrieving update information. Please try again later."

Additional messages from the Console Log:

     Error: An error occurred while parsing the update feed. (null) (URL (null))
     Error: Line 45: Opening and ending tag mismatch: link line 10 and head (null) (URL (null))

I read all the other posts that I could find online regarding this topic, and while Andy was able to resolve them, none of those issues are present in my case.  The only other thing that did not work (but should have according to the documentation) was ./bin/generate_appcast ... in which it complained about my app not having a Code Signature, gave an internal error and then dumped an incomplete appcast file.

So, I followed the directions of manually generating the appcast file (using ./bin/generate_keys which worked fine) to include a valid DSA Signature.  Of course, I updated the Info.plist file as per the documentation with the appropriate key-string pairs, and added the dsa_pub.pem file to the app bundle Resources.

And, I am using https from my server.  I also tried signing the dmg file (like the instructions for zip), but that didn't work either.

So, I am at a loss as to why I am still getting this error.  I was using a previous version of Sparkle (on an older Mac OSX), which worked just fine, but I need to upgrade to this new version for security reasons.  Only now, the new Sparkle version just doesn't work for us anymore, at least not on Mac OS High Sierra. I don't have another system to test it on at the moment.

Any suggestions of how to proceed?  Is this a known issue?  Help?

Below is my appcast file:

```
<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0" xmlns:sparkle="http://www.andymatuschak.org/xml-namespaces/sparkle" xmlns:dc="http://purl.org/dc/elements/1.1/">
<channel>
    <title>MyTestApp</title>
    <item>
        <title>Version 1.1</title>
        <pubDate>Tue, 29 May 2018 14:47:50 -0700</pubDate>
        <enclosure url="https://subpixeltech.com/TestApp/updates/TestApp.dmg" sparkle:version="1.1" length="4429830" type="application/octet-stream"  sparkle:dsaSignature="longStringOfCharactersRemovedForPostingHere" />
        <sparkle:minimumSystemVersion>10.7</sparkle:minimumSystemVersion>
    </item>
</channel>
</rss>
```

**Comment by Dave-K9** [comment]
My appcast file was cut off from my post, so let me try again.

<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0" xmlns:sparkle="http://www.andymatuschak.org/xml-namespaces/sparkle" xmlns:dc="http://purl.org/dc/elements/1.1/">
<channel>
    <title>MyTestApp</title>
    <item>
        <title>Version 1.1</title>
        <pubDate>Tue, 29 May 2018 14:47:50 -0700</pubDate>
        <enclosure url="https://subpixeltech.com/MyTestApp/updates/MyTestApp.dmg" sparkle:version="1.1" length="4429830" type="application/octet-stream"  sparkle:dsaSignature="longStringOfCharactersRemoved" />
        <sparkle:minimumSystemVersion>10.7</sparkle:minimumSystemVersion>
    </item>
</channel>
</rss>

**Comment by Dave-K9** [comment]
I had trouble pasting the contents of my appcast file (it gets cut off).  So, I will
[SubPixelRawMac-sav.xml.txt](https://github.com/sparkle-project/Sparkle/files/2052057/SubPixelRawMac-sav.xml.txt)

 try to upload it as a file.

**Comment by kornelski** [comment]
When you paste code in GitHub, surround it by ` ``` ` lines.

If the appcast seems fine, but Sparkle complains about invalid XML, it's likely that `SUFeedURL` in your `Info.plist` has a typo and the server is giving Sparkle an error page or some other file.

It's possible that we've accidentally added dependency on code signing. These days apps without it are rare, so this case doesn't get tested.

**Comment by Dave-K9** [comment]
OK, I bit the bullet and codesign'ed everything with an Apple Developer ID.  The latest Sparkle now works on High Sierra, although there is a problem in the bin/generate_appcast code that I will document in another thread.

**Comment by kornelski** [comment]
Great!
